vmf/scripts/mods/vmf/modules/core/packages.lua

```diff
@@ -0,0 +1,135 @@
+local vmf = get_mod("VMF")
+
+local _initialized = false
+
+local _mod_handles = {}
+local _queued_packages = {}
+local _loading_package = nil
+local _loaded_packages = {}
+
+function VMFMod:load_package(package_name, callback)
+  if not _initialized then
+    self:error("Package manager has not been initialized yet. It can only be used after the `all_mods_loaded` event has been processed.")
+    return
+  elseif self:has_package_loaded(package_name) then
+    self:error("Package '%s' has already been loaded", package_name)
+    return
+  end
+
+  if not _loaded_packages[self] then
+    _loaded_packages[self] = {}
+  end
+
+  local package_handle = string.format("resource_packages/%s/%s", self:get_name(), package_name)
+  local workshop_id = self:get_internal_data("workshop_id")
+
+  local resource_package = Mod.resource_package(_mod_handles[workshop_id], package_handle)
```

Comment:
What would happen if the `package_handle` is invalid?
Also, I think an invalid `workshop_id` should be handled too.

Reply:
`workshop_id` shouldn't be passed via `mod_data` at all. It's always correct when you grab it from vanilla mod_manager.